app/services/api_bridge/query.rb

```diff
@@ -113,8 +113,21 @@ def prepare_options(options)
       # combine the incoming params with the app default options
       params_hash = params_to_hash(options)
       opts = @app_options.clone.merge(params_hash) { |key, app, req|
-        if app.is_a?(Array) && req.is_a?(Array)
-          app + req
+        if key == "f"
+          # repeated filters are treated as ANDs and for that
```

Comment:
This is a little hard to follow, sort of written as run-on sentences.

I suggest something like this
```
# Remove app filter if request contains the same filter.
# Otherwise multiples of a filter are AND-ed in the API query
# and will likely return no results. To preserve any app filters,
# they must be sent along with the request filters.
```

I think you could remove the comment from inside the `reject` block too with this description. I think it covers what we're doing and why pretty well :)